public/scripts/d3/d3.js

```diff
@@ -1 +1,130 @@
-//  Add d3 functions here
+d3.json(echo.margins.jsonUrl, (err, data) => {
+  data.sort((a, b) => d3.descending(a.avgRating, b.avgRating));
+
+  let uniqueTags = [];
+  data.forEach((d) => {
+    uniqueTags.push(d.tag);
+  });
+  uniqueTags = uniqueTags.filter(onlyUnique);
```

Comment:
See my notes in the helper file.

But you should be accessing this function with `echo.helpers.onlyUnique`